File: src/main/java/com/metjka/vort/ui/VortApplication.java
package com.metjka.vort.ui;

import javafx.application.Application;
import javafx.scene.Scene;
import javafx.scene.text.Font;
import javafx.stage.Stage;

import java.util.prefs.Preferences;

/**
 * VortApplication application class for the GUI.
 */
public class VortApplication extends Application {
    /** A reference to the main window */
    private static Stage primaryStage;

    @Override
    public void start(Stage stage) throws Exception {
        primaryStage = stage;

        Font.loadFont(this.getClass().getResourceAsStream("/ui/fonts/titillium.otf"), 20);

        ToplevelPane toplevelPane = new ToplevelPane();
        MainOverlay overlay = new MainOverlay(toplevelPane);
        Scene scene = new Scene(overlay);
        Preferences prefs = Preferences.userNodeForPackage(VortApplication.class);
        String backGroundImage = prefs.get("background", "/ui/grid.png");
        toplevelPane.setStyle("-fx-background-image: url('" + backGroundImage + "');");
        String theme = prefs.get("theme", "/ui/colours.css");
        scene.getStylesheets().addAll("/ui/layout.css", theme);

        stage.setWidth(1024);
        stage.setHeight(768);

        stage.setOnCloseRequest(event -> System.exit(0));
        stage.setScene(scene);

        stage.show();
        toplevelPane.requestFocus();
        
    }

    /**
     * Get the main stage for this app
     *
     * @return the current stage
     */
    public static Stage getStage() {
        return primaryStage;
    }

    /**
     * @param args The command line arguments.
     */
    public static void main(String[] args) {
        launch(args);
    }
}

File: src/main/java/io/metjka/vortex/ui/VortApplication.kt
package io.metjka.vortex.ui

import io.metjka.vortex.ui.blocks.MathBlock
import javafx.application.Application
import javafx.fxml.FXMLLoader
import javafx.scene.Node
import javafx.scene.Scene
import javafx.scene.text.Font
import javafx.stage.Stage
import java.io.IOException
import java.util.prefs.Preferences

/**
 * Created by Ihor Salnikov on 24.2.2017, 9:53 PM.
 * https://github.com/metjka/VORT
 */
class VortApplication : Application() {

    companion object {

        var stagee: Stage? = null

        @JvmStatic
        fun main(args: Array<String>) {
            launch(VortApplication::class.java, *args)
        }
    }

    override fun start(primaryStage: Stage?) {

        stagee = primaryStage

        Font.loadFont(this::class.java.getResourceAsStream("/ui/fonts/FiraCode-Light.otf"), 20.0)

        val toplevelPane = TopLevelPane()
        val mainOverlay = MainOverlay(toplevelPane)
        val scene = Scene(mainOverlay)
        val prefs = Preferences.userNodeForPackage(VortApplication::class.java)
        val background = prefs.get("background", "/ui/grid.png")
        toplevelPane.style = "-fx-background-image: url('$background');"
        val theme = prefs.get("theme", "/ui/colours.css")
        scene.stylesheets.addAll("/ui/layout.css", theme)

        primaryStage?.width = 1024.0
        primaryStage?.height = 768.0
        primaryStage?.scene = scene
        primaryStage?.setOnCloseRequest { System.exit(0) }

        primaryStage?.show()

        toplevelPane.requestFocus()

        toplevelPane.addBlock(MathBlock(toplevelPane))
        initApp(toplevelPane)
    }

    fun initApp(toplevelPane: TopLevelPane) {
        toplevelPane.addBlock(MathBlock(toplevelPane))
    }
}


fun Node.loadXML(name: String) {

    try {
        val url = javaClass.getResource(String.format("/ui/%s.fxml", name))
        val fxmlLoader = FXMLLoader(url)
        fxmlLoader.setRoot(this)
        fxmlLoader.setController(this)

        fxmlLoader.load<Any>()
    } catch (e: IOException) {
        throw RuntimeException("A required FXML file, $name, could not be loaded.", e)
    }


}